lib/ecto/adapters/mysql/connection.ex

```diff
@@ -527,7 +527,7 @@ if Code.ensure_loaded?(Mariaex) do
     defp create_names(prefix, sources, pos, limit) when pos < limit do
       current =
         case elem(sources, pos) do
-          {table, schema} ->
+          %Ecto.Query.FromExpr{source: table, schema: schema} ->
```

Comment:
I think this is wrong. When we first create the sources, which I believe is in the planner, we should go ahead and store only the `{table, schema}` there. Making it consistent with Joins and others.

Reply:
Think as the sources tuple as containing already unstructured source expressions.